MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/DataProcessorUI/QDataProcessorWidget.h

```diff
@@ -86,7 +86,7 @@ class EXPORT_OPT_MANTIDQT_MANTIDWIDGETS QDataProcessorWidget
                             const std::string &defaultValue) override;
   bool askUserYesNo(std::string prompt, std::string title) override;
   void giveUserWarning(std::string prompt, std::string title) override;
-  void giveUserCritical(std::string prompt, std::string title) override;
+  void giveUserCritical(QString prompt, std::string title) override;
```

Comment:
From what I understand of the DataProcessorUI, the idea is to be only dependent on `Qt` where it is necessary. Hence the interfaces such as `DataProcessorView` (which is the interface for `QDataProcessorWidget`) do not depend on `Qt`. This was also true until recently for the `GenericDataProcessorWidget` (however it gained a `Qt` dependency through the async work; essentially it needs signals and slots).

Previously, the `QDataProcessorWidget` used `std::string` exclusively which was changed in parts to `QString` when Raquel implemented the Python exports, since `SIP` requires `QString`. 

We have two choices:
1. Give up the flexibility of easily using other UI frameworks, but make it use `Qt` structures consistently.
2. Remain with the flexibility, but be terribly inconsistent and make use of `QString::fromStdString` in a lot of places.

I would opt for `1.` since the core element of the `DataProcessorUI` is already tied up with `Qt` and having consistency would be a big improvement in terms of maintainability.